pkg/common/json.go

```diff
@@ -123,14 +127,58 @@ func UnmashalJSONResponse(resp *http.Response, v interface{}, b *[]byte) error {
 }
 
 func ReplaceJSONPlaceholders(source string, jsonData string) string {
-	replaced := source
-	regex := regexp.MustCompile("{([^}]*)}")
-	matches := regex.FindAllStringSubmatch(source, -1)
-	for _, selector := range matches {
-		value := gjson.Get(jsonData, selector[1]).String()
-		replaced = strings.ReplaceAll(replaced, "{"+selector[1]+"}", value)
+	var replaced, buffer []byte
+	var escaping, insidePlaceholder bool
+	var nestedCurlyBraces int
+
+	for _, b := range []byte(source) {
```

Comment:
Too bad there is no library to use to parse closing matching curly braces and support escaping them. Using regexp would be impossible?